src/resources/resources.controller.spec.ts

```diff
@@ -1,25 +1,271 @@
 import { Test, TestingModule } from "@nestjs/testing";
 import { ResourcesController } from "./resources.controller";
 import { ResourcesService } from "./resources.service";
+import { CreateResourceDto } from "./dto/create-resource.dto";
+import { CustomRequest } from "../global/types/CustomRequest";
+import { BadRequestException, NotFoundException } from "@nestjs/common";
+import { UpdateResourceDto } from "./dto/update-resource.dto";
 
 describe("ResourcesController", () => {
     let controller: ResourcesController;
 
+    const requestMock = {} as unknown as CustomRequest;
+    const dtoCreateMock = {
+        url: "https://chingu.com",
+        title: "Chingu",
+    } as CreateResourceDto;
+
+    // dto for updating a resource url
+    const dtoUpdateMock = {
+        url: "https://chingu-2.com",
+    } as UpdateResourceDto;
+    const mockTeamId: number = 1;
+
+    const mockResource = {
+        id: 1,
+        url: "https://chingu.com",
+        title: "Chingu",
+        teamMemberId: 1,
+        createdAt: new Date(Date.now()),
+        updatedAt: new Date(Date.now()),
+    };
+
+    const mockUpdatedResource = {
+        id: 1,
+        url: "https://chingu-2.com",
+        title: "Chingu",
+        teamMemberId: 1,
+        createdAt: new Date(Date.now()),
+        updatedAt: new Date(Date.now()),
+    };
+
+    const mockResourceArr = [
+        {
+            id: 1,
+            url: "https://chingu.com",
+            title: "Chingu",
+            teamMemberId: 1,
+            addedBy: {
+                member: {
+                    firstName: "Larry",
+                    lastName: "Castro",
+                    id: "18093ad0-88ef-4bcd-bee8-322749c876bd",
+                    avatar: "https://gravatar.com/avatar/90383a4ee0fb891c1ec3374e6a593a6c6fd88166d4fd45f796dabeaba7af836d?s=200&r=g&d=wavatar\n",
+                },
+            },
+            createdAt: new Date(Date.now()),
+            updatedAt: new Date(Date.now()),
+        },
+        {
+            id: 2,
+            url: "https://Nestjs.com",
+            title: "Nestjs",
+            teamMemberId: 2,
+            addedBy: {
+                member: {
+                    firstName: "John",
+                    lastName: "Doe",
+                    id: "18093ad0-88ef-4bcd-bee8-322749c876bd",
+                    avatar: "https://gravatar.com/avatar/90383a4ee0fb891c1ec3374e6a593a6c6fd88166d4fd45f796dabeaba7af836d?s=200&r=g&d=wavatar\n",
+                },
+            },
+            createdAt: new Date(Date.now()),
+            updatedAt: new Date(Date.now()),
+        },
+    ];
+
+    const mockResourcesService = {
+        createNewResource: jest.fn(),
+        findAllResources: jest.fn(),
+        updateResource: jest.fn(),
+        removeResource: jest.fn(),
+    };
```

Comment:
Extract the mock objects and variables to a section above the describe in the test file. This will help in read ability of the tests.